services/notifications.go

```diff
@@ -2528,48 +2528,61 @@ func (n *taxReportNotification) GetInfoMarkdown() string {
 func collectTaxReportNotificationNotifications(notificationsByUserID map[uint64]map[types.EventName][]types.Notification, eventName types.EventName) error {
 	tNow := time.Now()
 	firstDayOfMonth := time.Date(tNow.Year(), tNow.Month(), 1, 0, 0, 0, 0, time.UTC)
-	if tNow.Year() == firstDayOfMonth.Year() && tNow.Month() == firstDayOfMonth.Month() && tNow.Day() == firstDayOfMonth.Day() { // Send the reports on the first day of the month
-		var dbResult []struct {
-			SubscriptionID  uint64         `db:"id"`
-			UserID          uint64         `db:"user_id"`
-			Epoch           uint64         `db:"created_epoch"`
-			EventFilter     string         `db:"event_filter"`
-			UnsubscribeHash sql.NullString `db:"unsubscribe_hash"`
-		}
 
-		name := string(eventName)
-		if utils.Config.Chain.Config.ConfigName != "" {
-			name = utils.Config.Chain.Config.ConfigName + ":" + name
-		}
+	if tNow.Year() != firstDayOfMonth.Year() || tNow.Month() != firstDayOfMonth.Month() || tNow.Day() != firstDayOfMonth.Day() { // Send the reports on the first day of the month
+		return nil
+	}
 
-		err := db.FrontendWriterDB.Select(&dbResult, `
+	lastStatsDay, err := db.GetLastExportedStatisticDay()
+	if err != nil {
+		return err
+	}
```

Comment:
If the statistics export is delayed for a whole day, the user rewards for this month will never be sent because of the if above in line 2532. Is that correct?

If so, this should be addressed.

Reply:
Yes.

I implemented your requested change. This also "fixes" the weird behaviour that if the subscription was created on the first of the month the user would immediately get sent the report of the previous month.
Assuming nothing further along in the code stops that.